src/sentry/integrations/github/integration.py

```diff
@@ -833,38 +890,43 @@ def get_app_url(self) -> str:
 
     def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase:
         with record_event(IntegrationPipelineViewType.GITHUB_INSTALLATION).capture() as lifecycle:
+            self.active_user_organization = determine_active_organization(request)
+
+            if self.active_user_organization is not None and features.has(
+                "organizations:github-multi-org",
+                organization=self.active_user_organization.organization,
+                actor=request.user,
+            ):
+                error_page = self.check_pending_integration_deletion(request=request)
+                if error_page is not None:
+                    lifecycle.record_failure(GitHubInstallationError.PENDING_DELETION)
+                    return error_page
+
+                chosen_installation_id = pipeline.fetch_state("chosen_installation")
+                if chosen_installation_id is not None:
+                    pipeline.bind_state("installation_id", chosen_installation_id)
+                    return pipeline.next_step()
+
```

Comment:
is it possible to combine this new logic with the existing one in a way to avoid duplicate code as much as possible? both of them call `check_pending_integration_deletion` and i think we'd want to check that the `Integration` exists for both cases

Reply:
it's also possible we check that the list of installations exist as `Integrations` in our db before we even render them 🤔